src/sql/base/browser/ui/table/table.ts

```diff
@@ -180,7 +180,10 @@ export class Table<T extends Slick.SlickData> extends Widget implements IDisposa
 			this._data = new TableDataView<T>(data);
 		}
 		this._grid.setData(this._data, true);
-		this._data.filter(this._grid.getColumns());
+		if (!(data instanceof AsyncDataProvider)) {
+			// AsyncDataProvider does not support filtering currently.
```

Comment:
I'd prefer putting this in a try/catch and logging the error if it fails. We should never block viewing the data even if filtering failed